sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/DataBytesReceiver.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.fn.data;
+
+import org.apache.beam.model.fnexecution.v1.BeamFnApi;
+
+/**
+ * A receiver of elements in the form of a stream of bytes for an instruction, provided as
+ * {@link BeamFnApi.Elements.Data} messages.
+ */
+public interface DataBytesReceiver {
```

Comment:
In this case since we are using a strict and narrow type parameter on the consumer, do we need to create this interface type.

I understood for FnDataReceiver since that was using a generic type parameter that we wanted such a thing.

Reply:
It's java 7 - we don't have useful access to the java `Consumer` type.

Reply:
Ditto on revert because we can use JDK 8 now?